include/wintls/detail/sspi_handshake.hpp

```diff
@@ -19,13 +19,61 @@
 
 #include <wintls/handshake_type.hpp>
 
+#include <SubAuth.h>
+
 #include <array>
 #include <memory>
 #include <string>
 
 namespace wintls {
 namespace detail {
 
+typedef enum _eTlsAlgorithmUsage
+{
+    TlsParametersCngAlgUsageKeyExchange,
+    TlsParametersCngAlgUsageSignature,
+    TlsParametersCngAlgUsageCipher,
+    TlsParametersCngAlgUsageDigest,
+    TlsParametersCngAlgUsageCertSig
+} eTlsAlgorithmUsage;
+
+typedef struct _CRYPTO_SETTINGS
+{
+    eTlsAlgorithmUsage  eAlgorithmUsage;
+    UNICODE_STRING      strCngAlgId;
+    DWORD               cChainingModes;
+    PUNICODE_STRING     rgstrChainingModes;
+    DWORD               dwMinBitLength;
+    DWORD               dwMaxBitLength;
+} CRYPTO_SETTINGS, * PCRYPTO_SETTINGS;
+
+typedef struct _TLS_PARAMETERS
+{
+    DWORD               cAlpnIds;
+    PUNICODE_STRING     rgstrAlpnIds;
+    DWORD               grbitDisabledProtocols;
+    DWORD               cDisabledCrypto;
+    PCRYPTO_SETTINGS    pDisabledCrypto;
+    DWORD               dwFlags;
+} TLS_PARAMETERS, * PTLS_PARAMETERS;
+
+typedef struct _SCH_CREDENTIALS
+{
+    DWORD               dwVersion;
+    DWORD               dwCredFormat;
+    DWORD               cCreds;
+    PCCERT_CONTEXT* paCred;
+    HCERTSTORE          hRootStore;
+
+    DWORD               cMappers;
+    struct _HMAPPER **aphMappers;
+
+    DWORD               dwSessionLifespan;
+    DWORD               dwFlags;
+    DWORD               cTlsParameters;
+    PTLS_PARAMETERS     pTlsParameters;
+} SCH_CREDENTIALS, * PSCH_CREDENTIALS;
+
```

Comment:
Why are all these C types needed here? Aren't they already defined in some of the SSPI headers already included.

If they are really needed they definitely shouldn't be here.

Reply:
> Minimum supported client	Windows 10 1809 [desktop apps only]

Doing so helps us to compile on lower versions of Windows.

Reply:
> > Minimum supported client	Windows 10 1809 [desktop apps only]
> 
> Doing so helps us to compile on lower versions of Windows.

OK. Then please move it to it's own header and only include if compiling on a version of Windows where it's required. This should probably be in `sspi_types.h`.